src/common/oclengine.cpp

```diff
@@ -94,18 +103,27 @@ void OCLEngine::InitOCL()
     for (int i = 0; i < deviceCount; i++) {
         // a context is like a "runtime link" to the device and platform;
         // i.e. communication is possible
-        all_device_contexts.push_back(std::make_shared<OCLDeviceContext>(devPlatVec[i], all_devices[i]));
-        all_device_contexts[i]->context = cl::Context(all_devices[i]);
-        all_device_contexts[i]->queue = cl::CommandQueue(all_device_contexts[i]->context, all_devices[i]);
+        std::shared_ptr<OCLDeviceContext> devCntxt = std::make_shared<OCLDeviceContext>(devPlatVec[i], all_devices[i]);
+        devCntxt->context = cl::Context(all_devices[i]);
```

Comment:
Consider pushing these into the constructor instead of extending the initialization here.

Reply:
Done.